common/analysis/lint_waiver_config.lex

```diff
@@ -0,0 +1,98 @@
+%{
+#define _WAIVER_FLEXLEXER_H_
+#include <iostream>
+#include "common/analysis/lint_waiver_config.h"
+%}
+
+%option nodefault
+%option noyywrap
+%option prefix="verible"
+%option c++
+%option yyclass="verible::LintWaiverConfig"
+
+/* white space */
+LineTerminator \r|\n|\r\n
+InputCharacter [^\r\n\0]
+Space [ \t\f\b]
+
+DecimalDigits [0-9]+
+
+Alpha [a-zA-Z]
+RuleStart {Alpha}
+Letter {RuleStart}|"$"
+
+RuleName {RuleStart}({Letter}|-)*
+
+
+RuleParam -rule
+LineParam -line
+
+LineNumber {DecimalDigits}
+LineNumbers {DecimalDigits}{Space}+{DecimalDigits}
+
+WaiveCommand waive
+
+%x WAIVE_COMMAND
+%x WAIVE_RULE
+%x WAIVE_LINES
+
+%%
+
+{WaiveCommand} {
+  yy_push_state(WAIVE_COMMAND);
+}
+
+<WAIVE_COMMAND>{RuleParam} {
+  yy_push_state(WAIVE_RULE);
+}
+<WAIVE_RULE>{RuleName} {
+  yy_pop_state();
+
+  current_rule = absl::string_view(nullptr, 0);
+
+  for (auto r : active_rules) {
+    if (std::string(yytext) == r) {
```

Comment:
`absl::string_view(yytext, yyleng)` will avoid a `std::string` copy, and still be compare-able.